server/service/endpoint_utils.go

```diff
@@ -213,8 +213,15 @@ func NewUserAuthenticatedEndpointer(svc fleet.Service, opts []kithttp.ServerOpti
 	return &UserAuthEndpointer{svc: svc, opts: opts, r: r}
 }
 
+var pathReplacer = strings.NewReplacer(
+	"/", "_",
+	"{", "_",
+	"}", "_",
+)
+
 func getNameFromPathAndVerb(verb, path string) string {
-	return strings.ToLower(verb) + "_" + strings.ReplaceAll(strings.TrimSuffix("/api/v1/fleet/", strings.TrimRight(path, "/")), "/", "_")
+	return strings.ToLower(verb) + "_" +
+		pathReplacer.Replace(strings.TrimPrefix(strings.TrimRight(path, "/"), "/api/v1/fleet/"))
```

Comment:
Noticed this issue when adding the metrics output test, couldn't get a matching count of metrics/handler names vs requests made, was a bit of a head scratcher until I figured out the cause! :D That was because all new-format names were something like `get__api_v1_fleet_`.

Reply:
Yes! I found the same issue in the versioning PR. I guess not a lot of people are using prometheus?